go/vt/tabletmanager/binlog_test.go

```diff
@@ -0,0 +1,302 @@
+package tabletmanager
+
+import (
+	"flag"
+	"fmt"
+	"strings"
+	"testing"
+	"time"
+
+	"golang.org/x/net/context"
+
+	mproto "github.com/youtube/vitess/go/mysql/proto"
+	"github.com/youtube/vitess/go/sqltypes"
+	"github.com/youtube/vitess/go/vt/binlog/binlogplayer"
+	"github.com/youtube/vitess/go/vt/binlog/proto"
+	"github.com/youtube/vitess/go/vt/key"
+	"github.com/youtube/vitess/go/vt/mysqlctl"
+	myproto "github.com/youtube/vitess/go/vt/mysqlctl/proto"
+	"github.com/youtube/vitess/go/vt/topotools"
+	"github.com/youtube/vitess/go/vt/zktopo"
+
+	pb "github.com/youtube/vitess/go/vt/proto/topodata"
+)
+
+// fakeBinlogClient implements binlogplayer.Client
+type fakeBinlogClient struct {
+	t               *testing.T
+	expectedDialUID uint32
+
+	keyRangeChannel chan *proto.BinlogTransaction
+}
+
+func newFakeBinlogClient(t *testing.T, expectedDialUID uint32) *fakeBinlogClient {
+	return &fakeBinlogClient{
+		t:               t,
+		expectedDialUID: expectedDialUID,
+
+		keyRangeChannel: make(chan *proto.BinlogTransaction),
+	}
+}
+
+// Dial is part of the binlogplayer.Client interface
+func (fbc *fakeBinlogClient) Dial(endPoint *pb.EndPoint, connTimeout time.Duration) error {
+	if fbc.expectedDialUID != endPoint.Uid {
+		fbc.t.Errorf("fakeBinlogClient.Dial expected uid %v got %v", fbc.expectedDialUID, endPoint.Uid)
+	}
+	return nil
+}
+
+// Close is part of the binlogplayer.Client interface
+func (fbc *fakeBinlogClient) Close() {
+}
+
+// Dial is part of the binlogplayer.Client interface
+func (fbc *fakeBinlogClient) ServeUpdateStream(ctx context.Context, position string) (chan *proto.StreamEvent, binlogplayer.ErrFunc, error) {
+	return nil, nil, fmt.Errorf("NYI")
+}
+
+// Dial is part of the binlogplayer.Client interface
+func (fbc *fakeBinlogClient) StreamTables(ctx context.Context, position string, tables []string, charset *mproto.Charset) (chan *proto.BinlogTransaction, binlogplayer.ErrFunc, error) {
+	return nil, nil, fmt.Errorf("NYI")
+}
+
+// Dial is part of the binlogplayer.Client interface
+func (fbc *fakeBinlogClient) StreamKeyRange(ctx context.Context, position string, keyspaceIDType key.KeyspaceIdType, keyRange *pb.KeyRange, charset *mproto.Charset) (chan *proto.BinlogTransaction, binlogplayer.ErrFunc, error) {
+	c := make(chan *proto.BinlogTransaction)
+	var finalErr error
+	go func() {
+		for {
+			select {
+			case bt := <-fbc.keyRangeChannel:
+				c <- bt
+			case <-ctx.Done():
+				finalErr = ctx.Err()
+				close(c)
+				return
+			}
+		}
+	}()
+	return c, func() error {
+		return finalErr
+	}, nil
+}
+
+func TestBinlogPlayerMapHorizontalSplit(t *testing.T) {
```

Comment:
I agree that using functions will make this more readable.
I am adding another test to do vertical split, will factor out some code after that.